script.js

```diff
@@ -1,4 +1,251 @@
+  var score = 0;
+  for (var counter = 0; counter < handArray.length; counter += 1) {
+    score = score + getCardScore(handArray[counter]);
+  }
+  //if there are aces, update ace value in score. if total hand value is >21, update ace value from 11 to 1
+  for (var counter = 0; counter < handArray.length; counter += 1) {
+    if (handArray[counter].name === "ace") {
+      score = updateAceValueinScore(score);
+    }
+  }
+  return score;
+}
+
+function getCardScore(card) {
+  score = 0;
+  if (card.name === "king" || card.name === "queen" || card.name === "jack") {
+    score = 10;
+  } else if (card.name === "ace") {
+    score = 11;
+  } else {
+    score = card.rank;
+  }
+  return score;
+}
+
+function updateAceValueinScore(score) {
+  //if total value of hand is >21, update ace value from 11 to 1
+  if (score > 21) {
+    score = score - 10;
+  }
+  return score;
+}
+
+function displayPlayerHandAndScore(player, handArray, score, gameMode) {
+  message = "";
+
+  if (gameMode === "startGame") {
+    message = `${player}'s Hand: ${listCards(player, handArray, gameMode)}`;
+  } else {
+    message = `${player}'s Hand: ${listCards(
+      player,
+      handArray,
+      gameMode
+    )} Score: ${score}`;
+  }
+
+  if (isBust(handArray)) {
+    message = message + " (BUST!)";
+  }
+  message = message + "<br><br>";
+  return message;
+}
+
+function listCards(player, handArray, gameMode) {
+  var cardList = "";
+  if (player === "Dealer" && gameMode === "startGame") {
+    cardList =
+      cardList +
+      `- ${handArray[0].name} of ${handArray[0].suit}<br> - (Hidden Card)<br>`;
+  } else {
+    for (i = 0; i < handArray.length; i += 1) {
+      cardList =
+        cardList + `- ${handArray[i].name} of ${handArray[i].suit}<br>`;
+    }
+  }
+  return `<br> ${cardList}`;
+}
+
+function isBust(handArray) {
+  score = calculateScore(handArray);
+  return score > 21;
+}
+
+function generateResult(playerHandArray, dealerHandArray) {
+  playerScore = calculateScore(playerHandArray);
+  dealerScore = calculateScore(dealerHandArray);
+  gameResult = "";
+  if (isBust(dealerHandArray) && isBust(playerHandArray)) {
+    gameResult = "Tie!";
+  } else if (isBust(dealerHandArray)) {
+    gameResult = "Player wins!";
+  } else if (isBust(playerHandArray)) {
+    gameResult = "Dealer wins!";
+  } else if (playerScore == dealerScore) {
+    gameResult = "Tie!";
+  } else if (dealerScore > playerScore) {
+    gameResult = "Dealer  wins!";
+  } else {
+    gameResult = "Player wins!";
+  }
+  return gameResult;
+}
+
+//--- MAIN FUNCTION ---
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = "";
+
+  if (gameMode === "startGame") {
+    //game set up: create deck, shuffle deck and deal two cards to player and dealer.
+    dealerHand = [];
+    playerHand = [];
+    cardDeck = makeDeck();
+    cardDeck = shuffleCards(cardDeck);
+
+    dealerHand = dealCards(dealerHand, cardDeck, 2);
```

Comment:
It's alittle odd that you line 180 reassigns a value but line 181 does not. I understand that both work because its you're updating a global variable but you might want to just stick with one.